educator/src/Dscp/Educator/Web/Types.hs

```diff
@@ -204,3 +206,40 @@ deriving instance FromHttpApiData IsGraded
 deriving instance ToHttpApiData IsEnrolled
 deriving instance ToHttpApiData IsFinal
 deriving instance ToHttpApiData IsGraded
+
+---------------------------------------------------------------------------
+-- SQL instances
+---------------------------------------------------------------------------
+
+deriving instance HasSqlValueSyntax PgValueSyntax IsGraded
+deriving instance HasSqlEqualityCheck PgExpressionSyntax IsGraded
+
+---------------------------------------------------------------------------
+-- Filtering parameters
+---------------------------------------------------------------------------
+
+type IdentifierFilterTypes = [FilterMatching, FilterComparing]
+
+type instance SupportedFilters IsEnrolled = SupportedFilters Bool
+type instance SupportedFilters Course = IdentifierFilterTypes
+type instance SupportedFilters (Hash a) = SupportedFilters ByteString
+type instance SupportedFilters Address = DatetimeFilterTypes
+type instance SupportedFilters Timestamp = DatetimeFilterTypes
+type instance SupportedFilters ItemDesc = TextFilterTypes
+
+type instance FilteringParamTypesOf StudentInfo =
+    '[ "course" ?: 'ManualFilter Course ]
```

Comment:
Shouldn't this have an `isEnrolled` filter as well?

Reply:
I'm not sure how to fit this flag here. `StudentInfo` is returned by `GET students` endpoints, and basically, they accept a course against which we could check enrollment flag only like query parameter, and this "course" query parameter is not specified then the semantics of "isEnrolled" flag gets unclear.

Reply:
Isn't it possible to ignore `isEnrolled` when `course` is not specified?
Maybe also give it a default value so it is not necessary

Reply:
Which semantics would it have in couple with `course` query param?
If `isEnrolled = True` then only students enrolled to course `course` are returned, and if `isEnrolled = False` then such students are excluded?

Reply:
oh ok, now I get it. It doesn't make sense if the only filtering is on course enrollment anyway.

One question, because I don't remember if this is the case: is a student that completes a course still considered "enrolled" in it or not?
Because if it isn't then the this filter could have a use after all.

Reply:
That makes a point. Currently, a student is unrolled even if he completes the course, although I'm not sure whether does it make sense :)
When we polish the business logic, I will keep this filtering use case in mind.